libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorMapper.kt

```diff
@@ -328,6 +334,67 @@ internal class NavigatorMapper {
                     .alertGeometry(alert.getAlertGeometry())
                     .build()
             }
+            RouteAlertType.KINCIDENT -> {
+                val incidentInfo = alert.incidentInfo!!
```

Comment:
Let's avoid hard cast and null-check properly.

Reply:
In other cases (like `KSERVICE_AREA`), we're just not appending the type if it's not present, we're not throwing. We should align the approach.

Reply:
> In other cases (like `KSERVICE_AREA`), we're just not appending the type if it's not present, we're not throwing. We should align the approach.

I agree ☝️

Reply:
@LukasPaczos @Guardiola31337 here's need trade off, because https://github.com/mapbox/mapbox-navigation-android/pull/3640#discussion_r502515215 `id` is required for builder. it must be presented when we have incident.

Reply:
For consistency I think we should design this following the same approach as with other alerts, for instance `TunnelInfo` or `CountryBorderCrossingAdminInfo`.

Reply:
Something around these lines 👀 

```kotlin
IncidentAlert.Builder(
                    alert.beginCoordinate,
                    alert.distance
                )
                    .alertGeometry(alert.getAlertGeometry())
                    .info(alert.incidentInfo?.toIncidentInfo())
                    .build()
```

Reply:
done